### PR TITLE
adrienne / fix cloudflare deployment security

### PR DESCRIPTION
**Pull request by adrienne-deriv**
Changes:

-  CF Deployment for test links

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .github/workflows/generate-preview-link.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            - name: Download artifact
+              id: download-artifact
+              uses: dawidd6/action-download-artifact@v2
+              with:
+                  workflow_conclusion: success
+                  run_id: ${{ github.event.workflow_run.id }}
+                  name: 'pr-${{ github.event.workflow_run.id }}'
+
+            - name: Retrieve and verify user organization
+              id: pr_information
```
Comment (adrienne-deriv) [inline .github/workflows/generate-preview-link.yml:40]: This checks if the user is from binary-com organization, thus ensuring only binary-com user's PR can perform the build and deploy steps

File: .github/workflows/generate-preview-link.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+              run: |
+                  echo "Verifying user's organization..."
+                  user=$(cat ./USERNAME)
+                  response=$(curl -s -L \
+                      -w "%{http_code}" \
+                     -o /dev/null -H "Accept: application/vnd.github+json" \
+                     -H "Authorization: Bearer ${{ secrets.LIST_ORGS_TOKEN }}" \
```
Comment (adrienne-deriv) [inline .github/workflows/generate-preview-link.yml:47]: We need a separate Github classic token for this with SSO verification for binary-com organization
cc: @balakrishna-deriv

File: .github/workflows/generate-preview-link.yml (hunk 1 above)
Comment (adrienne-deriv) [reply .github/workflows/generate-preview-link.yml:40]: Worst case scenario, all PRs outside binary-com can be blocked and approved for workflow run first:
Screenshot 2023-09-22 at 4 02 17 PM

File: .github/workflows/generate-preview-link.yml (hunk 2 above)
Comment (balakrishna-deriv) [reply .github/workflows/generate-preview-link.yml:47]: Will github.token work? I think it might work because we are only doing a read operation.

Comment (adrienne-deriv) [reply .github/workflows/generate-preview-link.yml:47]: Noo it needs a new token that is SSOed with binary-com organization, the GITHUB_TOKEN is provided by default

Comment (adrienne-deriv) [reply .github/workflows/generate-preview-link.yml:47]: Screenshot 2023-09-25 at 12 14 51 PM

**Review by balakrishna-deriv — APPROVED** [review]

**Review by habib-deriv — DISMISSED** [review]
LGTM

**Review by yashim-deriv — COMMENTED** [review]
LGTM

**Review by yashim-deriv — APPROVED** [review]

**Comment by balakrishna-deriv** [comment]
LGTM

**Pull request merged**